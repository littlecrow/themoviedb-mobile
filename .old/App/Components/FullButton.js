import React, { Component } from 'react';
import { TouchableOpacity, Text } from 'react-native';
import styles from './Styles/FullButtonStyles';
import PropTypes from 'prop-types';

// Note that this file (App/Components/FullButton) needs to be
// imported in your app somewhere, otherwise your component won't be
// compiled and added to the examples dev screen.

// Ignore in coverage report
/* istanbul ignore next */

export default class FullButton extends Component {
  static propTypes = {
    text: PropTypes.string,
    onPress: PropTypes.func,
    styles: PropTypes.object
  }

  render () {
    return (
      <TouchableOpacity style={[styles.button, this.props.styles]} onPress={this.props.onPress}>
        <Text style={styles.buttonText}>{this.props.text && this.props.text.toUpperCase()}</Text>
      </TouchableOpacity>
    );
  }
}
