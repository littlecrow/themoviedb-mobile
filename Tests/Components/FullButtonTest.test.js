import 'react-native';
import React from 'react';
import FullButton from '../../App/Components/FullButton';

import renderer from 'react-test-renderer';

test('FullButton component renders correctly', () => {
  const _getComponent = (props) => {
    return (
      renderer.create(
        <FullButton text={props.text} />
      ).toJSON()
    );
  };

  let component = _getComponent({ text: 'Hi' });
  expect(component).toMatchSnapshot();
});
