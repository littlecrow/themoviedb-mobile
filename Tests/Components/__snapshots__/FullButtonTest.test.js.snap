// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullButton component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#2ecc71",
      "borderBottomColor": "#27ae60",
      "borderBottomWidth": 1,
      "borderTopColor": "#2ecc71",
      "borderTopWidth": 1,
      "height": 40,
      "marginVertical": 5,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ecf0f1",
          "fontSize": 14,
          "fontWeight": "bold",
          "margin": 15,
          "marginLeft": 0,
        }
      }
    >
      HI
    </Text>
  </View>
</View>
`;
